bpf/lib/conntrack.h

```diff
@@ -499,7 +499,9 @@ ipv4_extract_tuple(struct __ctx_buff *ctx, struct ipv4_ct_tuple *tuple,
 	tuple->nexthdr = ip4->protocol;
 
 	if (unlikely(tuple->nexthdr != IPPROTO_TCP &&
-		     tuple->nexthdr != IPPROTO_UDP))
+		     tuple->nexthdr != IPPROTO_UDP &&
+		     tuple->nexthdr != IPPROTO_IPIP))
```

Comment:
Without this change, the IPIP packets will be dropped  before handle_dsr_v4.

Reply:
Did you test that? I see `ipv4_extract_tuple` is called from `extract_tuple4`, which is itself called only from `ctx_redirect_to_proxy_first`. `ctx_redirect_to_proxy_first` is called only from `bpf_host`'s `to-host` section, only in case of a packet going to the proxy. Am I missing something?

Reply:
I see. I did not test this change alone, but tested it with another change. I will remove it and test again. Thanks.
btw, I saw other comments about adding new test, updating doc, refactoring a function etc. I will try to address all of them.

Reply:
Awesome! Don't hesitate to ask for help if needed. I'm pchaigno on Slack as well.